File: py/9465.py
"""
1
5
50 10 100 20 40
30 50  70 10 60
"""
import sys
dp = [ [0 for x in range(100005)], [0 for x in range(100005)] ]
sys.setrecursionlimit(100005)

def f(i, j):
  if j >= n-1:
    return arr[i][n-1]
  nxti = 0
  if i == 0:
    nxti = 1
  if dp[i][j]==0:
    dp[i][j] = max(f(nxti, j+1), f(nxti, j+2)) + arr[i][j]
  return dp[i][j]

a = int(input())
for i in range(a):
  n = int(input())
  arr = []
  for j in range(n):
    dp[0][j] = 0
    dp[1][j] = 0
  arr.append(list(map(int, sys.stdin.readline().split())))
  arr.append(list(map(int, sys.stdin.readline().split())))

  print(max(f(0, 0), f(1, 0)))

File: cpp/9465.cpp


#include <cstdio>
#include <cstring>
int dp[2][100001] = {0};
int arr[2][100001] = {0};
int n;
int m;
int f(int i, int j) {
  if (j >= m - 1) {
    return arr[i][m-1];
  }
  if (dp[i][j] < 0) {
    int nxti = i ? 0 : 1;
    int a = f(nxti, j+1);
    int b = f(nxti, j+2);
    dp[i][j] = (a > b ? a : b) + arr[i][j];
  }
  return dp[i][j];
}
int main(void) {
  scanf("%d", &n);
  for (int i = 0;i<n;i++) {
    scanf("%d", &m);
    for (int j=0;j<m;j++){
      scanf("%d", &arr[0][j]);
      dp[0][j]=-1;
      dp[1][j]=-1;
    }
    for (int j=0;j<m;j++){
      scanf("%d", &arr[1][j]);
      
    }
    int a = f(0, 0);
    int b = f(1, 0);
    printf("%d\n", a>b?a:b);
  }
  return 0;
}